### PR TITLE
#4690 - Prompt default users to change their password on login

### LINKED ISSUE #4690: Prompt default users to change their password on login
### Situation Description
During server setup, server administrators can configure SORMAS to generate a number of default users that are useful for testing and development purposes. However, these users also have default passwords and thus are a potential security breach, especially if administrators forget to disable these users on production systems.

Default user generation is turned off by default, but there's still a mandatory admin user with a standard password generated.

### Feature Description
The new feature should present a prompt at login to any of the standard users using their standard password (as is defined in `StartupShutdownService:createDefaultUsers`) to change their password. The password changing logic can be the same as when editing a user from the user directory.

### PR DESCRIPTION
**Pull request by schwzr**
# What does this PR do?

- This PR fixes #4690.
- It provides a new feature to present a prompt at login to any of the standard users with their standard password (as defined in `StartupShutdownService:createDefaultUsers`) to change their password. 
- If we have an elevated user capable of editing other users, this user can change the standard password of other standard users as well. The new passwords can be exported as CSV.
- The attached screencast demonstrates this new feature:

sormas-4690

We are happy to receive feedback from your side!

Thanks to @Snorzze for providing the unit tests!

### CODE REVIEW

File: sormas-api/src/main/java/de/symeda/sormas/api/utils/DefaultUserHelper.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+	public static boolean usesDefaultPassword(String username, String passwordHash, String seed) {
+		return getDefaultPassword(username).filter(s -> passwordHash.equals(PasswordHelper.encodePassword(s, seed))).isPresent();
+	}
+
+	public static boolean currentUserUsesDefaultPassword(List<UserDto> allUsersWithDefaultPassword, UserDto currentUser) {
+		return allUsersWithDefaultPassword.contains(currentUser);
+	}
+
+	public static boolean otherUsersWithDefaultPassword(List<UserDto> allUsersWithDefaultPassword, UserDto currentUser) {
```
Comment (MateStrysewske) [inline sormas-api/src/main/java/de/symeda/sormas/api/utils/DefaultUserHelper.java:84]: Rename to `otherUsersUseDefaultPassword`

File: sormas-ui/src/main/java/de/symeda/sormas/ui/login/DefaultPasswordOtherScreen.java [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class DefaultPasswordOtherScreen extends Window {
```
Comment (MateStrysewske) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/login/DefaultPasswordOtherScreen.java:51]: Rename to `ChangeDefaultPasswordsWindow`

File: sormas-ui/src/main/java/de/symeda/sormas/ui/login/DefaultPasswordOwnScreen.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+import de.symeda.sormas.api.FacadeProvider;
+import de.symeda.sormas.api.i18n.Captions;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.i18n.Strings;
+import de.symeda.sormas.api.user.UserDto;
+import de.symeda.sormas.ui.utils.ButtonHelper;
+import de.symeda.sormas.ui.utils.CssStyles;
+
+public class DefaultPasswordOwnScreen extends Window {
```
Comment (MateStrysewske) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/login/DefaultPasswordOwnScreen.java:35]: Rename to `ChangeDefaultUserPasswordWindow`

**Review by MateStrysewske — CHANGES_REQUESTED** [review]
Hey! Thanks for your contribution! Seems like your IDE settings aren't quite correct when it comes to import orders. java imports are supposed to be above the symeda imports. Please make sure that your IDE is set up correctly and re-format your files.

**Comment by schwzr** [comment]
Thanks @MateStrysewske. I updated my IDE settings to not revert the import ordering of the Eclipse Code Formatting Plugin automatically and renamed the classes and method names as you requested.

**Review by MateStrysewske — APPROVED** [review]
Thank you! Android indeed supports some Java 8 features, but it's still using Java 7 at its core, and it especially requires external code resources (just like our API) to be written in Java 7, unfortunately.

Also, for future reference if you want to contribute code to this repository again - please make sure that every commit always references the issue number (not just the first one). That's important for us to ensure a clean commit history. I've overseen it myself when I did my initial review, so it's alright for now. :)

**Comment by rzo1** [comment]
@MateStrysewske Thanks for your time reviewing our pull request. For future contribution, we will adhere to the contribution guidelines. May I ask, if this will be included in the upcoming release?

**Comment by MateStrysewske** [comment]
@rzo1 Yes, this feature will be part of 1.59.0!

**Pull request merged**